Rubberduck.CodeAnalysis/Inspections/Abstract/ImplicitSheetReferenceInspectionBase.cs

```diff
@@ -35,7 +36,8 @@ protected override IEnumerable<Declaration> ObjectionableDeclarations(Declaratio
 
         private static readonly string[] GlobalObjectClassNames =
         {
-            "Global", "_Global"
+            "Global", "_Global", 
+            "Worksheet", "_Worksheet"
```

Comment:
This feels wrong. 1) we should be able to enumerate document modules already, and infer the base class without needing to check a list of constants. As it is, it won’t port to other hosts. 2) by definition, the appobjects are predeclared, but appobjects are different in other hosts. We can then infer Excel’s Globals because it’s an appobject.

To avoid needing resolving the interface to a coclass using strings we should enumerate coclasses’ implemented interfaces and thus locate the correct coclass.

Reply:
This inspection is very much Excel-specific though. Agreed the inspection feels like it's upside-down and needs adjustment.

Reply:
Looking at this again, I see the code is actually in inspection - I was conflating that with the `Parsing` project. Still, I think it should be the `Parsing` project that provides the necessary metadata for the inspection to then work upon. the inspection can/should filter for specific Excel coclasses/interfaces, but it should be doing off a list of some kind, not a bunch of constant strings.

Reply:
Absolutely: with properly-resolving unqualified `Workbook` and `Worksheet` members, it should be possible to identify implicit `ActiveWorkbook` and implicit `ActiveSheet` references by simply pulling the identifier references of `_Global`.